### PR TITLE
Jelauria first deliverables

### PR DESCRIPTION
**Pull request by jelauria**
Added the following:
`jelauria_Sprocs.sql`: Stored Procedures to generate new pronoun set and mark shift as worked
`jelauria_ComputedColumns.sql`: computing `NumCurrentCounselors` & `SpentCompensatingRAs`
`jelauria_BusinessRules.sql`: defines rules `MustBePaidExtraOnXmas` and `NoViolatingAvailablility`
`jelauria_LookUpTables.sql`: populates `tblPHONE_TYPE` and `tblLOCATION_TYPE`

### CODE REVIEW

File: jelauria_ComputedColumns.sql [hunk 1]
```diff
@@ -0,0 +1,44 @@
+-- Computed column calculating the number of current counselors by gender
+CREATE FUNCTION jelauria_CALC_NumCurrentCounselorsByGender(@PK INT)
+RETURNS INT
+AS
+BEGIN
+    DECLARE @RET int
+    SET @RET = (SELECT COUNT(tEP.EmpPosID)
```
Comment (andreybutenko) [inline jelauria_ComputedColumns.sql]: Should this be `COUNT(DISTINCT tEP.EmpPosID)`?

File: jelauria_ComputedColumns.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        FROM tblGENDER tG
+            JOIN tblEMPLOYEE tE on tG.GenderID = tE.GenderID
+            JOIN tblEMPLOYEE_POSITION tEP on tE.EmployeeID = tEP.EmployeeID
+            JOIN tblPOSITION tP on tEP.PositionID = tP.PositionID
+            JOIN tblPOSITION_TYPE tPT on tP.PositionTypeID = tPT.PositionTypeID
+        WHERE tPT.PositionTypeName = 'Counselor'
+            AND tEP.EndDate IS NULL
+            AND tG.GenderID = @PK)
+END
```
Comment (andreybutenko) [inline jelauria_ComputedColumns.sql:17]: You have not actually returned anything :open_mouth:

File: jelauria_ComputedColumns.sql [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            JOIN tblCOMPENSATION tC on tS.CompID = tC.CompID
+            JOIN tblEMP_SHIFT_STATUS tESS on tS.ShiftID = tESS.ShiftID
+            JOIN tblEMPLOYEE_POSITION tEP on tESS.EmpPosID = tEP.EmpPosID
+            JOIN tblPOSITION tP on tEP.PositionID = tP.PositionID
+            JOIN tblPOSITION_TYPE tPT on tP.PositionTypeID = tPT.PositionTypeID
+            JOIN tblSTATUS tST on tESS.StatusID = tST.ShiftStatusID
+        WHERE tST.StatusTitle = 'Worked'
+            AND tPT.PositionTypeName = 'RA'
+            AND tQ.QuarterID = @PK)
+END
```
Comment (andreybutenko) [inline jelauria_ComputedColumns.sql:42]: Did not return any value

File: jelauria_ComputedColumns.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        FROM tblQUARTER tQ
+            JOIN tblSHIFT tS on tQ.QuarterID = tS.QuarterID
+            JOIN tblCOMPENSATION tC on tS.CompID = tC.CompID
+            JOIN tblEMP_SHIFT_STATUS tESS on tS.ShiftID = tESS.ShiftID
+            JOIN tblEMPLOYEE_POSITION tEP on tESS.EmpPosID = tEP.EmpPosID
+            JOIN tblPOSITION tP on tEP.PositionID = tP.PositionID
+            JOIN tblPOSITION_TYPE tPT on tP.PositionTypeID = tPT.PositionTypeID
+            JOIN tblSTATUS tST on tESS.StatusID = tST.ShiftStatusID
+        WHERE tST.StatusTitle = 'Worked'
+            AND tPT.PositionTypeName = 'RA'
```
Comment (andreybutenko) [inline jelauria_ComputedColumns.sql:39]: I don't think this is necessary, consider excluding it to be flexible of other position types which may get compensation

**Review by andreybutenko — CHANGES_REQUESTED** [review]
Great work!

File: jelauria_BuinessRules.sql [hunk 5]
```diff
@@ -0,0 +1,57 @@
+-- Business Rule: No employee can be assigned a shift if they have stated they are unavailable for it
```
Comment (acohen29) [inline jelauria_BuinessRules.sql:1]: All looks solid, we will have to see how they work when pushed to the db

File: jelauria_BuinessRules.sql [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            JOIN tblMONTH tM on tS.MonthID = tM.MonthID
+            JOIN tblDAY tD on tS.DayID = tD.DayID
+            JOIN tblCOMPENSATION tC on tS.CompID = tC.CompID
+        WHERE tM.[MonthName] = 'December'
+            AND (tD.[DayName] = '24' OR tD.[DayName] = '25')
+            AND tC.CompAmount <= 0
+    )
+    BEGIN 
+       SET @RET = 1
+    END
```
Comment (acohen29) [inline jelauria_BuinessRules.sql]: Just needs a 'Return @Ret'

File: jelauria_BuinessRules.sql (hunk 5 above)
Comment (acohen29) [reply jelauria_BuinessRules.sql:1]: Just make sure to handle the return

File: jelauria_ComputedColumns.sql (hunk 4 above)
Comment (jelauria) [reply jelauria_ComputedColumns.sql:39]: did this to join to the shift_type table just for the sake of having another join/filter parameter; will remain unchanged

File: jelauria_LookUpTables.sql [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+INSERT INTO tblPHONE_TYPE(PhoneTypeID, PhoneTypeTitle, PhoneTypeDescr)
+VALUES (1, 'RA Mobile', 'Personal mobile phone to contact a current RA working at the location'),
+       (2, 'Office', 'Landline phone at the residential life office for the location'),
+       (3, 'CA Mobile', 'Personal phone to contact the CA for a given region'),
+       (4, 'Counselor', 'Phone number to contact a counselor for crisis situations at a campus'),
+       (5, 'RD Mobile', 'Phone number to contact a current RD or CM for a given location')
+GO
+
+-- Populates tbl LOCATION TYPE
+INSERT INTO tblLOCATION_TYPE(LocationTypeID, LocationName, LocationDescription)
```
Comment (andreybutenko) [inline jelauria_LookUpTables.sql:11]: `LocationName` -> `LocationTypeName`

**Review by andreybutenko — CHANGES_REQUESTED** [review]

File: jelauria_ComputedColumns.sql (hunk 1 above)
Comment (jelauria) [reply jelauria_ComputedColumns.sql]: done!

**Comment by jelauria** [comment]
all commented edits have been made

**Pull request merged**